references/cuegen/tag_test.go

```diff
@@ -0,0 +1,193 @@
+		{"json_cue_4", `json:",inline" cue:"default:default_value"`, &tagOptions{Name: "", Inline: true, Default: str("default_value"), Enum: defaultEnum}},
+		{"json_cue_5", `json:"name,omitempty,inline" cue:"default:default_value"`, &tagOptions{Name: "name", Optional: true, Inline: true, Default: str("default_value"), Enum: defaultEnum}},
+		{"json_cue_6", `json:",omitempty,inline" cue:"default:default_value;enum:enum1,enum2"`, &tagOptions{Name: "", Optional: true, Inline: true, Default: str("default_value"), Enum: []string{"enum1", "enum2"}}},
+	}
+
+	g := &Generator{}
+	for _, tt := range tests {
+		got := g.parseTag(tt.tag)
+		assert.Equal(t, tt.opts, got, tt.name)
+	}
+}
+
+func TestParseBasicTag(t *testing.T) {
+	tests := []struct {
+		name     string
+		tag      string
+		wantName string
+		wantOpt  string
+	}{
+		{"empty", "", "", ""},
+		{"only_name", "name", "name", ""},
+		{"name_and_opt", "name,opt,opt2", "name", "opt,opt2"},
+		{"name_and_opt_2", "name,opt1;opt2.opt3", "name", "opt1;opt2.opt3"},
+		{"name_and_opt_with_space", "name, opt", "name", " opt"},
+		{"only_opt", ",opt,opt2", "", "opt,opt2"},
+	}
+
+	for _, tt := range tests {
+		gotName, gotOpt := parseTag(tt.tag)
+		assert.EqualValues(t, tt.wantName, gotName, tt.name)
+		assert.EqualValues(t, tt.wantOpt, gotOpt, tt.name)
+	}
+}
+
+func TestBasicTagOptHas(t *testing.T) {
+	tests := []struct {
+		name string
+		tag  string
+		opt  string
+		want bool
+	}{
+		{"empty", "", "", false},
+		{"empty_opt", "name", "", false},
+		{"empty_tag", "", "opt", false},
+		{"single_opt", "name,opt", "opt", true},
+		{"multi_opt", "name,opt1,opt2,opt3", "opt1", true},
+		{"multi_opt_2", "name,opt1,opt2,opt3", "opt2", true},
+		{"multi_opt_3", "name,opt1,opt2,opt3", "opt3", true},
+		{"only_multi_opt", ",opt1,opt2,opt3", "opt1", true},
+		{"only_multi_opt_2", ",opt1,opt2,opt3", "opt2", true},
+		{"only_multi_opt_3", ",opt1,opt2,opt3", "opt3", true},
+	}
+
+	for _, tt := range tests {
+		_, opt := parseTag(tt.tag)
+		assert.Equal(t, tt.want, opt.Has(tt.opt), tt.name)
+	}
+}
+
+func TestParseExtTag(t *testing.T) {
+	tests := []struct {
+		name string
+		tag  string
+		want map[string]string
+	}{
+		{"empty", "", map[string]string{}},
+		{"only_key", "key", map[string]string{"key": ""}},
+		{"one_kv", "key:value", map[string]string{"key": "value"}},
+		{"multi_kv", "key1:value1;key2:value2", map[string]string{"key1": "value1", "key2": "value2"}},
+		{"bool", "key1;key2", map[string]string{"key1": "", "key2": ""}},
+		{"bool_and_kv", "key1;key2:value2", map[string]string{"key1": "", "key2": "value2"}},
+		{"kv_and_bool", "key1:value1;key2", map[string]string{"key1": "value1", "key2": ""}},
+		{"multi_kv_and_bool", "key1:value1;key2:value2;key3", map[string]string{"key1": "value1", "key2": "value2", "key3": ""}},
+		{"escape", "key1:\"value\"", map[string]string{"key1": `"value"`}},
```

Comment:
Sorry but what's meaning here by "supporting escape" because I think 
```go
"key1:\"value\""
```
means `key1:"value"` literally. It's transformed by Golang and it's not your parsing logic to support the escaping. If there's any end to end test case about this feature can you point it?

Reply:
Sorry, it's mistake, I will remove this case. The `escape` in tag is used to escape the separator(`;`).

And does this feature require e2e testing? I'm not familiar with this, can you help me?

Reply:
Updated

Reply:
There's some similar test case marked relative to escaping. What do you mean by escape the separator? Can you add some suitable test case?

Reply:
If you want to use a separator(`;`) in the key or value, you need to use the escape, otherwise it will be treated as a new field.

Reply:
More tests added

Reply:
Got that.